src/eigen/diag.jl

```diff
@@ -28,18 +28,25 @@ function diagonalize_all_kblocks(eigensolver, ham::Hamiltonian, nev_per_kpoint::
                   "$nev_per_kpoint eigenvalues. Increase Ecut.")
         end
         # Get ψguessk
-        @timing "QR orthonormalization" begin
-            if ψguess !== nothing
-                # ψguess provided
+        if ψguess !== nothing
+            nev_guess = size(ψguess[ik], 2)
+            if nev_guess > nev_per_kpoint
+                ψguessk = ψguess[ik][:, 1:nev_per_kpoint]
+            elseif nev_guess == nev_per_kpoint
                 ψguessk = ψguess[ik]
-            elseif interpolate_kpoints && ik > 1
-                # use information from previous k-point
-                X0 = interpolate_kpoint(results[ik - 1].X, ham.basis, kpoints[ik - 1],
-                                        ham.basis, kpoints[ik])
-                ψguessk = ortho_qr(X0)  # Re-orthogonalize and renormalize
             else
-                ψguessk = random_orbitals(ham.basis, kpt, nev_per_kpoint)
+                X0 = similar(ψguess[ik], n_Gk, nev_per_kpoint)
+                X0[:, 1:nev_guess] = ψguess[ik]
```

Comment:
this was factored out to random_orbitals to be more flexible in the future (eg if we want to support symmetries of the wfs, where we possibly need to impose some specific symmetry on the initial orbitals)